packages/fx-core/src/plugins/resource/bot/v2/common.ts

```diff
@@ -47,7 +47,8 @@ export function decideTemplateScenarios(ctx: Context, inputs: Inputs): Set<strin
         const notificationTriggerType = (inputs[
           QuestionNames.BOT_HOST_TYPE_TRIGGER
         ] as string[]) ?? [AppServiceOptionItem.id];
-        notificationTriggerType.forEach((triggerType) => {
+        // notificationTriggerType may be string in VS scenario
```

Comment:
It looks a little weird that VS and VSCode return different type because they are the same question model and VS UI need to handle it as the correct type instead of asking plugin to workaround it.
We could write here as a workaround to unblock integration test, but it is better to ask Zhiyu to see if there is any missing from UI side and if he can have a better solution for it.

Reply:
So, when we support more notification trigger type in preview 3, we will revisit this logic and revert it back to previous one?